### DISCUSSION TITLE
Memory leak when OpenGlControl remove adn add to panel.

### DISCUSSION THREAD

**Discussion by Coloryr** [AvaloniaUI/Avalonia#13519] [Q&A] [answered]
```
public void Add(IUserControl con)
{
      Controls.Content = con;
}

public async void Close()
{
    Controls.Content = null;
}
```

And call `Add` and `Close` many times.

![image](https://github.com/AvaloniaUI/Avalonia/assets/26276037/8deb5b64-a40f-4c13-8703-eabc5a7742de)

[control xaml](https://github.com/Coloryr/ColorMC/blob/master/src/ColorMC.Gui/UI/Controls/Main/MainControl.axaml)

**Comment by Coloryr** [comment]
[报告20231107-1649.zip](https://github.com/AvaloniaUI/Avalonia/files/13277764/20231107-1649.zip)

**Comment by thevortexcloud** [comment]
Garbage collection in ,NET is not really deterministic. It happens when the garbage collector thinks it needs to happen. As such this may not even be an actual issue. You can prompt the GC to run, by doing `GC.Collect()`.

But it's possible there is a real memory leak here. I have not looked.

**Comment by Coloryr** [comment]
[src.zip](https://github.com/AvaloniaUI/Avalonia/files/13290734/src.zip)
run demo

**Comment by Coloryr** [comment]
Is OpenGlControlBase issus.
when window have OpenGlControl, it will leak.
[AvaloniaApplication1.zip](https://github.com/AvaloniaUI/Avalonia/files/13290817/AvaloniaApplication1.zip)

**Reply by timunie** [reply, marked as answer]
made an issue for now, as I have no idea what's wrong here.
